tests/translate/storage/test_php.py

```diff
@@ -316,6 +316,29 @@ def test_parsing_arrays_keys_with_spaces(self):
         assert phpunit.name == "$lang->'item 1'"
         assert phpunit.source == "value1"
 
+    def test_parsing_arrays_keys_with_quotes(self):
+        """Ensure that our identifiers get escaped. Bug #5348."""
+        phpsource = """return [
+         'item\\'s 1' => 'value1',
+         "item's 2" => 'value2',
+         'item "3"' => 'value3',
+         "item \\"4\\"" => 'value4',
+      ];"""
+        phpfile = self.phpparse(phpsource)
+        assert len(phpfile.units) == 4
+        phpunit = phpfile.units[0]
+        assert phpunit.name == "return[]->'item's 1'"
+        assert phpunit.source == "value1"
+        phpunit = phpfile.units[1]
+        assert phpunit.name == "return[]->'item's 2'"
+        assert phpunit.source == "value2"
+        phpunit = phpfile.units[2]
+        assert phpunit.name == "return[]->'item \"3\"'"
+        assert phpunit.source == "value3"
+        phpunit = phpfile.units[3]
+        assert phpunit.name == "return[]->'item \"4\"'"
+        assert phpunit.source == "value4"
```

Comment:
Reading through this again, it does test only parsing, not serialization, something like this will be needed for that:

```suggestion
        assert phpunit.source == "value4"
        
        assert bytes(phpfile).decode() == phpsource
```